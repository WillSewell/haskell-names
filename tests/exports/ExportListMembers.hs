{-# LANGUAGE MultiParamTypeClasses #-}

module ExportListMembers (Foo(Foo1, Foo3, c), Bar(x), N(N)) where

data Foo = Foo1 | Foo2 { c :: Bool } | Foo3 { d :: Bool }

class Bar where x :: Foo; y :: Foo

newtype N = N { unN :: Foo }

a = a

data Baz = Baz
