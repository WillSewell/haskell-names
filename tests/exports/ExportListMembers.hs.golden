Symbols
  (fromList
     [ SymMethod
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName = GName { gModule = "ExportListMembers" , gName = "x" }
               }
         , sv_fixity = Nothing
         , sv_className =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Bar" }
               }
         }
     , SymSelector
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName = GName { gModule = "ExportListMembers" , gName = "c" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo" }
               }
         , sv_constructors =
             [ OrigName
                 { origPackage = Nothing
                 , origGName =
                     GName { gModule = "ExportListMembers" , gName = "Foo2" }
                 }
             ]
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo1" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo" }
               }
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo3" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo" }
               }
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName = GName { gModule = "ExportListMembers" , gName = "N" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName = GName { gModule = "ExportListMembers" , gName = "N" }
               }
         }
     ])
  (fromList
     [ SymData
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Foo" }
               }
         , st_fixity = Nothing
         }
     , SymNewType
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName = GName { gModule = "ExportListMembers" , gName = "N" }
               }
         , st_fixity = Nothing
         }
     , SymClass
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListMembers" , gName = "Bar" }
               }
         , st_fixity = Nothing
         }
     ])